pkgs/profpatsch/nman/nman.rs

```diff
@@ -137,28 +138,65 @@ fn build_man_page<'a>(drv: DrvWithOutput, section: &str, page: &str, tempdir: &T
 
     let mut path = PathBuf::from(OsStr::from_bytes(first_path));
     path.push("share/man");
-    path.push(format!("man{}", section));
-    path.push(page);
-
-    path.set_extension(format!("{}.gz", section));
-
-    if path.exists() {
-        Ok(Some(path))
-    } else {
-        // check for uncompressed man page if the derivation
-        // has no / a custom fixupPhase
-        path.set_extension(section);
-
-        if path.exists() {
-            Ok(Some(path))
-        } else {
-            Ok(None)
+
+    // no share/man, no man pages
+    if !path.exists() {
+        return Ok(None);
+    }
+
+    // expected sub directory of share/man or, if no section
+    // is given, all potential sub directories
+    let mut section_dirs: Vec<OsString> =
+        match section {
+            Some(s) => vec!(OsString::from(format!("man{}", s))),
+            None => {
+                std::fs::read_dir(path.as_path())
+                    .map_err(|_| NmanError::IO)?
+                    .filter_map(|entry| entry.ok())
+                    .map(|entry| entry.file_name())
+                    .collect()
+            },
+        };
+
+    // sorting should be ascending in terms of numerics,
+    // apart from that, not many requirements
+    section_dirs.sort();
+
+    const EXTENSIONS: [&str; 2] = [ ".gz", "" ];
```

Comment:
I wouldn’t parse any extension tbh, maybe split off at the last `.` and match both with and without the dot. I don’t think manpages usually contain a `.` though.

Reply:
We don't parse the extension we append it. The intention is to figure out which the n possible filenames for our man page are and check if one of them exists. This is preferrable IMO over iterating through the directory and checking for every if file that is our man page.

Reply:
I think what we should be doing instead here is list the files and compare against the target, should be just one syscall anyway (and ofc not relevant compared to the instantiation overhead).

Reply:
Done in 4237940.